src/config/localfile-config.c

```diff
@@ -1479,13 +1479,25 @@ bool w_logreader_journald_merge(logreader ** logf_ptr, size_t src_index) {
     bool dst_has_filters = logr[dst_index].journal_log->filters != NULL
                            && logr[dst_index].journal_log->filters[0] != NULL;
 
-    // Disable filter is already disabled or if any don't have filters
-    if (!src_has_filters || !dst_has_filters) {
+    // Only disable filters if BOTH blocks have no filters
+    // If at least one has filters, we use the union of non-empty filters
+    if (!src_has_filters && !dst_has_filters) {
+        // Both have no filters - keep the "collect everything" behavior
```

Comment:
I am leaving the following suggestion because I believe it would simplify the conditional block. However, the current code is correct and functional.
Therefore, I leave it up to you to decide whether to apply the change:
```suggestion
    // Only disable filters if BOTH blocks have no filters
    // If at least one has filters, we use the union of non-empty filters
    bool should_disable_filters = (!src_has_filters && !dst_has_filters);
    logr[dst_index].journal_log->disable_filters = should_disable_filters;

    if (should_disable_filters) {
        mwarn(LOGCOLLECTOR_JOURNAL_CONFG_DISABLE_FILTER);
    }
```